src/qibocal/protocols/characterization/rabi/utils.py

```diff
@@ -190,3 +190,13 @@ def extract_rabi(data):
     if "RabiLength" in data.__class__.__name__:
         return "length", "Time [ns]", rabi_length_function
     raise RuntimeError("Data has to be a data structure of the Rabi routines.")
+
+
+def period_correction_factor(phase: float):
+    """Correct period by taking phase into account.
+
+    https://github.com/qiboteam/qibocal/issues/656
+    """
+    # solution of cos (2 pi x / period + phase) = +/- 1 for k in [-2,-1,0,1,2]
+    # with k that gets correction closest to period
+    return np.round(phase / np.pi + 1) - phase / np.pi
```

Comment:
```py
np.round(phase / np.pi + 1) == np.round(phase / np.pi) + 1
np.round(x) - x == - np.modf(x)[0]
```
I was sure there was a function for the fractional part in NumPy, but unfortunately it's a bit less explicit than that (since it's returning both the fractional and the integral part - most likely because it also computes the second to get the first, internally).

```suggestion
    return 1 - np.modf(phase / np.pi)[0]
```